### PR TITLE
Minor sceNpTrophy Fixes (déjà vu)

### PR DESCRIPTION
**Pull request by elad335**
* Don't writeback data when an error returned in sceNpTrophyGetTrophyInfo.
* Fix error code of invalid trophy ID in sceNpTrophyGetTrophyInfo.
* Fix off-by-one progress callbacks count in sceNpTrophyRegisterContext, substruct 1 from "total" value reported.
This means for a trophy config with 0 trophies progress callbacks won't be executed, instead of 1 callback in master.
* Fix arguments in the first callback of sceNpTrophyRegisterContext.

Testcase for these : https://github.com/elad335/myps3tests/tree/master/ppu_tests/Unregistered%20Trophy%20Context

### CODE REVIEW

File: rpcs3/Emu/Cell/Modules/sceNpTrophy.cpp [hunk 1]
```diff
@@ -601,7 +601,7 @@ error_code sceNpTrophyRegisterContext(ppu_thread& ppu, u32 context, u32 handle,
 		for (u32 old_value; current < until && (old_value = *queued);
 			current = get_system_time())
 		{
-			queued->wait(old_value, atomic_wait_timeout{std::min<u64>((until - current) * 1000, 300'000'000)});
```
Comment (elad335) [inline rpcs3/Emu/Cell/Modules/sceNpTrophy.cpp]: std::min was a no-op, I know because I added it.

File: rpcs3/Emu/Cell/Modules/sceNpTrophy.cpp [hunk 2]
```diff
@@ -601,7 +601,7 @@ error_code sceNpTrophyRegisterContext(ppu_thread& ppu, u32 context, u32 handle,
...
+			queued->wait(old_value, atomic_wait_timeout{(until - current) * 1000});
```
Comment (elad335) [inline rpcs3/Emu/Cell/Modules/sceNpTrophy.cpp]: std::min was a no-op, I know because I added it.

**Comment by elad335** [comment]
Fixed a typo I noticed in sceNpTrophyGetTrophyInfo on invalid trophy ID error code, which also made it fail to compile after remvoing not_an_error, lol.

**Comment by elad335** [comment]
Btw.. in the testcase there are exactly 11 trophies in config.

File: rpcs3/Emu/Cell/Modules/sceNpTrophy.cpp [hunk 3]
```diff
@@ -1042,7 +1040,17 @@ static error_code NpTrophyGetTrophyInfo(const trophy_context_t* ctxt, s32 trophy
 
 	if (!found)
 	{
-		return not_an_error(SCE_NP_TROPHY_INVALID_TROPHY_ID);
+		return SCE_NP_TROPHY_ERROR_INVALID_TROPHY_ID;
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/Modules/sceNpTrophy.cpp]: lmao 3 year old bug

**Comment by Megamouse** [comment]
yolo

**Pull request merged**